### PR TITLE
Parse all command line options in one place and within function

### PR DESCRIPTION
**Pull request by ashahba**
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/kubeflow/1942)

### CODE REVIEW

**Comment by ashahba** [comment]
The purpose of this PR is to address parsing all command line options in a single location as requested by @jlewi here:
https://github.com/kubeflow/kubeflow/pull/1888#discussion_r230949335

File: scripts/kfctl.sh [hunk 1]
```diff
@@ -1,4 +1,4 @@
-#!/bin/bash
+#!/usr/bin/env bash
```
Comment (jlewi) [inline scripts/kfctl.sh]: Why this change?

File: scripts/kfctl.sh [hunk 2]
```diff
@@ -157,9 +107,63 @@ createNamespace() {
...
+# Generate all required components
+customizeKsApp() {
+  ks param set ambassador platform ${KUBEFLOW_PLATFORM}
+  ks param set jupyter platform ${KUBEFLOW_PLATFORM}
+}
+
+ksApply() {
+  pushd ${KUBEFLOW_KS_DIR}
+
+  if [ "${PLATFORM}" == "minikube" ]; then
```
Comment (jlewi) [inline scripts/kfctl.sh]: Should we always just create the Namespace? 
createNamespace should be a null op if the namespace doesn't exist.

File: scripts/kfctl.sh [hunk 3]
```diff
@@ -157,9 +107,63 @@ createNamespace() {
...
+    fi
+  fi
+  set -x
+}
+
+
 if [ "${COMMAND}" == "init" ]; then
   DEPLOYMENT_NAME=${WHAT}
 
+  # Parse all command line options
```
Comment (jlewi) [inline scripts/kfctl.sh]: Should we move the code for argument parsing into a function 
parseArgs

and the remaining code into a function
named  main

and then just invoke main in the script?

**Comment by jlewi** [comment]
Thank you so much for doing this.
Just a couple nitpicky comments.

File: scripts/kfctl.sh (hunk 3 above)
Comment (ashahba) [reply scripts/kfctl.sh]: I think that'll make the code even cleaner.
I'll put those inside a function.

File: scripts/kfctl.sh (hunk 2 above)
Comment (ashahba) [reply scripts/kfctl.sh]: Agreed. I think the `if` is not needed here.

**Comment by ashahba** [comment]
@jlewi This one is ready to go let me know if anything else needs to be done.

File: scripts/kfctl.sh (hunk 1 above)
Comment (jlewi) [reply scripts/kfctl.sh]: Seems good; thanks for the explanation.

**Comment by jlewi** [comment]
Thanks.
/lgtm
/approve
